### PR TITLE
Hooks: hook files for issue 4209

### PR DESCRIPTION
**Pull request by strangemind**
hook-firestore.py not hook-google-cloud-firestore.py
hook-firebase_admin.py
hook-grpc.py

hook-firestore.py

from MartinSahlen/cloud-functions-python#56
from PyInstaller.utils.hooks import copy_metadata
datas = copy_metadata('google-cloud-core')
datas += copy_metadata('google-cloud-firestore')
datas += copy_metadata('google-api-core')

hook-firebase_admin.py
from PyInstaller.utils.hooks import copy_metadata
datas = copy_metadata('google-api-core')
datas += copy_metadata('firebase-admin')

hook-grpc.py

from stackoverflow.com/questions/55848884/google-cloud-firestore-doesnt-get-added-to-pyinstaller-build
from PyInstaller.utils.hooks import copy_metadata, collect_data_files

datas = collect_data_files('grpc')

needs all the resources not just the py files

### CODE REVIEW

File: PyInstaller/hooks/hook-firestore.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#tsd 5-1-19
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-firestore.py]: Please remove this line.

File: PyInstaller/hooks/hook-firestore.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# from https://github.com/MartinSahlen/cloud-functions-python/issues/56
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-firestore.py]: If you want to keep this reverence, please move it below the copyright header.

File: PyInstaller/hooks/hook-firestore.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#-----------------------------------------------------------------------------
+# Copyright (c) 2013-2018, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License with exception
+# for distributing bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+from PyInstaller.utils.hooks import copy_metadata
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-firestore.py]: Missing blank line above

File: PyInstaller/hooks/hook-firestore.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+datas = copy_metadata('google-cloud-core')
+datas += copy_metadata('google-cloud-firestore')
+datas += copy_metadata('google-api-core')
+
+#pythonhosted.org/pyinstaller/hooks.html#understanding-pyinstaller-hooks
+#
+
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-firestore.py]: Please remove the trailing comments and trailing blank lines.

File: PyInstaller/hooks/hook-grpc.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+# from stackoverflow.com/questions/55848884/google-cloud-firestore-doesnt-get-added-to-pyinstaller-build
+from PyInstaller.utils.hooks import copy_metadata, collect_data_files
+
+datas = collect_data_files('grpc')
+
+
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-grpc.py]: Please remove the trailing blank lines.

**Comment by htgoebel** [comment]
Thanks for for update. The [**changelog entry**](https://pyinstaller.readthedocs.io/en/latest/development/changelog-entries.html) are still missing. Please ad it to the respective commit. The PR should consist of *three* commits, one for each hook.

Please see https://pyinstaller.readthedocs.io/en/stable/development/commit-messages.html#please-write-good-commit-messages for why I am so picky.

When updating a pull-request, you can simply (force) push the updated branch to github again. This will automatically update the pull-request (which follows the branch, not the commit). So you do not need to close the pull-request and open a new one. This also has the benefit that the discussion history is kept. For detailed instructions please read [Updating a Pull-Request](https://pyinstaller.readthedocs.io/en/stable/development/pull-requests.html#updating-a-pull-request) in the manual.

**Comment by htgoebel** [comment]
additionally there are still some lint errors, see https://travis-ci.org/pyinstaller/pyinstaller/jobs/559473241

**Comment by htgoebel** [comment]
Thanks for the new update. but the changelog entries are still missing. Without these the pull-request wll not be merged. I'm neither writing these for you nor remember you again. Please provide the changelog entries soon as described above, otherwise I will close the pull-request.

**Comment by strangemind** [comment]
change log file placed in the branch.

**Comment by htgoebel** [comment]
@strangemind I have to admit that I'm quite angry about this pull-request. This is your second pull-request, which is just a mess. Sorry to say, but i you want to have this merged any time soon, please clean it up. I'm not spending any time on getting firebase into PyInstaller, as Firebase is a google survailance tool. If you want it in, clean up, so i jsut need to click "rebase and merge".

**Comment by Legorooj** [comment]
@strangemind status please

**Comment by strangemind** [comment]
The hook files work will for our company.  If others need the files for Google Firestore, they can read the issue log.    The confusion seems how to format the files so pyinstaller can merge the files.

**Pull request closed without merging**